packages/utils/test/schema/getClosestMatchingOptionTest.ts

```diff
@@ -167,6 +167,52 @@ export default function getClosestMatchingOptionTest(testValidator: TestValidato
       });
       expect(getClosestMatchingOption(testValidator, schema, formData, get(schema, 'items.oneOf'))).toEqual(1);
     });
+    it('returns the second option when data matches for anyOf', () => {
+      const schema: RJSFSchema = {
+        type: 'array',
+        items: {
+          anyOf: [
+            {
+              properties: {
+                lorem: {
+                  type: 'string',
+                },
+              },
+              required: ['lorem'],
+            },
+            {
+              properties: {
+                ipsum: {
+                  anyOf: [
+                    {
+                      properties: {
+                        day: {
+                          type: 'string',
+                        },
+                      },
+                    },
+                    {
+                      properties: {
+                        night: {
+                          type: 'string',
+                        },
+                      },
+                    },
+                  ],
+                },
+              },
+              required: ['ipsum'],
+            },
+          ],
+        },
+      };
+      const formData = { ipsum: { night: 'nicht' } };
+      // Mock to return true for the last of the second one-ofs
+      testValidator.setReturnValues({
+        isValid: [false, false, false, false, false, false, false, true],
+      });
```

Comment:
Wondering if in some places it might be more readable/maintainable if we stubbed the implementation to return `isValid` for a specific subschema (in this case the second anyOf) instead of relying on ordering.

Reply:
Most likely... What were you thinking of changing?